spock-core/src/main/java/org/spockframework/runtime/FeatureNode.java

```diff
@@ -10,6 +10,7 @@
 import static java.util.stream.Collectors.toSet;
 
 public abstract class FeatureNode extends SpockNode<FeatureInfo> {
+  private boolean alreadyPrepared = false;
```

Comment:
This will probably cause issues when parallel execution is enabled. I don't know if it will suffice to make it into an `AtomicBoolean`/`volatile`.

Reply:
I thought about that when writing it.
But I think any sane usage of such dependencies need to use `SAME_THREAD` anyway which the `@Stepwise` also already implies.
So I can add a `volatile` but I don't think that would change much.

Reply:
Then, shouldn't adding dependencies imply the `SAME_THREAD` execution mode on all? If we would add this new feature then we should minimize the potential for problems.
(No need to fix this at the moment until the other issues are solved, just wanted to point it out.)